### ISSUE TITLE
make format breaks raw strings

### ISSUE THREAD

**Issue by zwass** [facebook/osquery#1934]
`make format` will change raw strings (see http://en.cppreference.com/w/cpp/language/string_literal (6)) in a way that causes both compiler and logical errors.

Take the line:
```
  std::string json = R"({" foo ": " Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmo "})";
```

After `make format`, this becomes
```
  std::string json = R "({" foo ": " Lorem ipsum dolor sit amet,
              consectetur adipiscing elit, sed do eiusmo "})";
```

The space after the `R` results in a compiler error, and simply removing this space still leaves invalid JSON because the newlines are interpreted literally in a raw string. It's not particularly difficult to change this example back, but it's trickier with more complex JSON like in a unit test.

**Comment by zwass** [comment]
https://llvm.org/bugs/show_bug.cgi?id=20957 sheds some light on this, and indicates that it's probably our use of `Standard: Cpp03` in the `.clang-format` that is causing the problem (https://github.com/facebook/osquery/blob/master/.clang-format#L30).

Are there concerns with changing it to C++11?

**Comment by zwass** [comment]
fixed by #1935
